pkg/tar/tar.go

```diff
@@ -91,17 +91,21 @@ func (t *stiTar) SetExclusionPattern(p *regexp.Regexp) {
 	t.exclude = p
 }
 
-// StreamFileAsTar streams the source file as a tar archive.
+// StreamFileAsTar streams the source directory as a tar archive.
 // The permissions of the file is changed to 0666.
 func (t *stiTar) StreamDirAsTar(source, dest string, writer io.Writer) error {
 	f, err := os.Open(source)
 	if err != nil {
 		return err
 	}
-	if info, _ := f.Stat(); !info.IsDir() {
+	defer f.Close()
+	info, err := f.Stat()
+	if err != nil {
+		return err
```

Comment:
i think it's reasonable to make callers of StreamFileAsTar do the logging of the error if they so choose, which is the current behavior for errors on open as @php-coder noted anyway.